app/models/miq_action.rb

```diff
@@ -1018,4 +1024,22 @@ def invoke_or_queue(
       MiqQueue.put(args)
     end
   end
+
+  def target_hosts(action, rec)
+    if action.options[:use_event_target]
+      ipaddress(rec)
+    elsif action.options[:use_localhost]
+      'localhost'
+    else
+      action.options[:hosts]
+    end
+  end
+
+  def ipaddress(record)
+    record.ipaddresses[0] if record.respond_to?(:ipaddresses)
+  end
+
+  def target_user(record)
+    record.respond_to?(:tenant_identity) ? record.tenant_identity : User.find("admin")
```

Comment:
This method should be called `tenant_identity` and the "admin" condition should be following the pattern from here: https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/tenant_identity_mixin.rb#L8